### PR TITLE
185: add first campaign screen

### LINKED ISSUE #183: Campaign: Homepage Posters
**As an app user, I want to see all posters displayed on a map in oder to enter the location for a newly hung poster, to be able to repair/replace a damaged poster, or to remove a poster entirely (e.g. after an election has ended).**

The main map for the user to look at or manage posters. Hung posters, posters that need to be replaced and posters that have already been taken down are coordinated in an easily understandable map.

Registered poster locations are shown as icons on the map. Areas and Polling Stations [_Wahllokale_] entered into the separate campaign planning tool (until October 2024: _Wahlatlas_ / as of October 2024: in parallel development by Pollion) can be displayed and hidden using a filter.
Posters can be added through a click on the central add-POI element (address is entered analog to the Doors functionality and, optionally, with name and photo).
Poster icons are green (active), red (damaged), or gray (marked as taken down). Tapping on a poster icon shows the commensurate picture in an overlay, a tap on the overlay allows the user to edit (address inputs) and show / edit / remove the picture in full size. Furthermore, the poster can be set as damaged or taken down through toggles, as well as commented in a free text field.
Where needed, posters can be removed completely from the map. The button "_Eintrag löschen_" triggers the feedback: "_Möchtest du das Plakat wirklich - endgültig - löschen? Markiere abgehängte Plakate bitte als abgehängt, statt sie zu löschen!_" In the overlay "_Meine Plakate_", users can see a list of posters they have registered, which are sorted by date of entry. These posters can also be edited and deleted from this overlay.

Acceptance criteria:
- [x] #185 
- [x] #267 
- [ ] #216
- [ ] #279
- [ ] On the map, users can identify damaged poster locations (red dot)
- [ ] On the map, users can identify poster locations (green dot)
- [ ] On the map, users can identify poster locations that have been taken down (grey dot)
- [ ] #173 
- [ ] #197 
- [ ] #263 
- [ ] #269 
- [ ] #285 
- [ ] #286 
- [ ] #287 
- [ ] #288 
- [ ] #289 
- [ ] #290 

Design:
361200427-10d2c435-f6fa-4ead-9460-3e34d47e7b60
Prototype:
https://www.figma.com/proto/7wE0Iveas3GGb8u4oL0714/App-Design-Mitglieder-App-(1)-(Copy)?node-id=2670-9551&t=b8swD5IOAFfZ0OSo-1&starting-point-node-id=2425%3A17429

API endpoint reference: https://github.com/verdigado/gruene-api/issues/61

Pollion reference: https://github.com/pollion/sunflower-app/issues/25

### LINKED ISSUE #266: Campaign | Flyers: Submenu
As an app user I want to be able to click on Flyers in the menu bar #185 to open up a submenu for flyers. The submenu has a single option #270.

Acceptance criteria:
- [x] 1st position: "Routen" (no link)

Further information:
- the submenu option can be toggled on or off and is initially off.
- #270

Note: Menu updated with #342 

Bildschirmfoto 2024-11-03 um 13 52 04
Bildschirmfoto 2024-11-03 um 20 14 22

Prototype: https://www.figma.com/proto/7wE0Iveas3GGb8u4oL0714/App-Design-Mitglieder-App-(1)-(Copy)?node-id=2670-9749&t=b8swD5IOAFfZ0OSo-1&starting-point-node-id=2425%3A17429

API endpoint reference: https://github.com/verdigado/gruene-api/issues/82

Pollion reference: https://github.com/pollion/sunflower-app/issues/136

### LINKED ISSUE #267: Campaign | Posters: Submenu
As an app user, I want a submenu for posters with the following options. 

Acceptance criteria:
- [x] 1st position: "Routen" (no link in Release 0*) 
- [x] 2nd position: "Wahllokale" (no link in Release 0*)  

An updated version is found in #340 

Further info for later releases:
- the menu options can be toggled on or off and is initially off.
- #271 
- #220 

Bildschirmfoto 2024-11-03 um 13 51 54
Bildschirmfoto 2024-11-03 um 20 24 04

Prototype: https://www.figma.com/proto/7wE0Iveas3GGb8u4oL0714/App-Design-Mitglieder-App-(1)-(Copy)?node-id=2683-13221&t=b8swD5IOAFfZ0OSo-1&starting-point-node-id=2425%3A17429

Pollion reference: https://github.com/pollion/sunflower-app/issues/132

### LINKED ISSUE #268: Campaign | Doors: Submenu
Clicking on doors in the menu bar #185 opens up a submenu for doors. The submenu has three options.

Acceptance criteria:
- [x] 1st position: "Besuchte Gebiete" (no link)
- [x] 2nd position: "Routen (no link)
- [x] 3rd position: "Fokusgebiete" links to #169
- [x] menu option "Fokusgebiete" toggled OFF initially
- [x] the menu options can be toggled on or off 

Note: Updated in #338.

Further information for later releases:
- #272 
- menu option "Besuchte Gebiete" toggled ON initially
- #170
- menu options "Fokusgebiete" and "Routen" toggled OFF initially

Bildschirmfoto 2024-11-03 um 13 51 45
Bildschirmfoto 2024-11-03 um 20 28 42

Prototype: https://www.figma.com/proto/7wE0Iveas3GGb8u4oL0714/App-Design-Mitglieder-App-(1)-(Copy)?node-id=2685-14248&t=b8swD5IOAFfZ0OSo-1&starting-point-node-id=2425%3A17429

Pollion reference: https://github.com/pollion/sunflower-app/issues/128

### PR DESCRIPTION
**Pull request by Stift**
### Short description

Incorporate menu structures for campaign screen.

### Proposed changes

- new menu structure for campaign screen
- tabs for TEAMS and STATISTICS are disabled (as requested by PO)
- some sub-menu items are disabled (as requested by PO)
- menu items may differ from issue descriptions as these have not yet been updated (PO?)

### Side effects

- -

### Resolved issues

- fixes #183 
- fixes #266 
- fixes #267 
- fixes #268

### CODE REVIEW

File: lib/i18n/app_de.json [hunk 1]
```diff
@@ -1,53 +1,75 @@
...
+    "label": "Artikel"
+  },
+  "campaigns": {
+    "campaigns": "Wahlkampf",
+    "label": "Wahlkampf",
+    "filters": {
+      "focusAreas": "Fokusgebiete",
+      "routes": "Routen",
+      "visited_areas": "Besuchte Gebiete",
+      "polling_station": "Wahllokale",
```
Comment (Burtchen) [inline lib/i18n/app_de.json:16]: Let's keep the plural and say polling_station**s**

File: lib/i18n/app_de.json [hunk 2]
```diff
@@ -1,53 +1,75 @@
...
+      "experience_areas": "Erfahrungsgebiete"
+    },
+    "posters": {
+      "label": "Plakate"
+    },
+    "doors": {
+      "label": "Haustür"
```
Comment (Burtchen) [inline lib/i18n/app_de.json:23]: Plural?

File: lib/i18n/app_de.json [hunk 3]
```diff
@@ -1,53 +1,75 @@
...
+    },
+    "flyer": {
+      "label": "Flyer"
+    },
+    "team": {
+      "label": "Team"
+    },
+    "statistics": {
+      "label": "Statistik"
```
Comment (Burtchen) [inline lib/i18n/app_de.json:32]: Plural?

**Review by Burtchen — APPROVED** [review]
I have some string bike-shedding, but otherwise woop woop 🚢

File: lib/app/theme/theme.dart [hunk 4]
```diff
@@ -1,104 +1,129 @@
...
+
+  static TextStyle mediumText = GoogleFonts.ptSans(
+    textStyle: TextStyle(
+      fontSize: 18,
+      fontWeight: FontWeight.w400,
+      height: 1.3,
+      letterSpacing: 0.05,
+    ),
+  );
+}
```
Comment (steffenkleinle) [inline lib/app/theme/theme.dart:90]: Can we stick to the flutter theme naming/reuse an existing text style here?

**Review by steffenkleinle — APPROVED** [review]
Nice :tada: Would be nice if you could still adjust the linebreaks to `LF` (currently using `CRLF`)

**Pull request merged**